api/environments/identities/traits/models.py

```diff
@@ -3,14 +3,31 @@
 from django.core.exceptions import ObjectDoesNotExist
 from django.db import models
 from django.utils.encoding import python_2_unicode_compatible
-from simple_history.models import HistoricalRecords
+from django_lifecycle import AFTER_DELETE, AFTER_SAVE, LifecycleModel, hook
 
 from environments.identities.traits.exceptions import TraitPersistenceError
 from environments.models import BOOLEAN, FLOAT, INTEGER, STRING
 
 
+class TriatQuerySet(models.QuerySet):
+    def delete(self, *args, **kwargs):
+        # collect all the identities that we need to update
+        identitie_to_delete = [trait.identity for trait in self]
+        super().delete(*args, **kwargs)
+        # update all the identities
+        if identitie_to_delete:
+            identitie_to_delete[0].bulk_send_to_dynamodb(identitie_to_delete)
```

Comment:
This is bad. Why are we calling a static method against an instantiated object of the class? You can simplify these 2 lines to and (if you need to) handle the empty list in the `bulk_send_to_dynamo` method. 

```python
Identity.bulk_send_to_dynamo(...)
```